### PR TITLE
tasks.c:2067:11: warning: this condition has identical branches [-Wduplicated-branches] #476

### PR DESCRIPTION
**Pull request by pierrenoel-bouteville-act**
Description
-----------
Replace duplicated if branch because empty by a comment to avoid warning/error with option GCC -Wduplicated-branches

Test Steps
-----------
Build file with -Wduplicated-branches and -Werror with GCC

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: include/task.h [hunk 1]
```diff
@@ -1683,6 +1683,16 @@ configSTACK_DEPTH_TYPE uxTaskGetStackHighWaterMark2( TaskHandle_t xTask ) PRIVIL
                                         uint32_t * pulIdleTaskStackSize ); /*lint !e526 Symbol not defined as it is an application callback. */
 #endif
 
+#if ( configUSE_MALLOC_FAILED_HOOK == 1 )
```
Comment (aggarg) [inline include/task.h:1686]: This change seems unrelated?

Comment (pierrenoel-bouteville-act) [reply include/task.h:1686]: Sorry both pull request was done, last commit should be in other pull request.

Comment (aggarg) [reply include/task.h:1686]: Can you remove these changes from this PR?

File: tasks.c [hunk 2]
```diff
@@ -2064,15 +2064,12 @@ void vTaskStartScheduler( void )
...
-        else
-        {
-            /* Should only reach here if a task calls xTaskEndScheduler(). */
-        }
+        xPortStartScheduler();
+        /* if just above call of xPortStartScheduler not return pdFALSE
+         *      Should not reach here as if the scheduler is running the
+         *      function will not return.
+         * otherwise
+         *      Should only reach here if a task calls xTaskEndScheduler(). */
```
Comment (aggarg) [inline tasks.c:2072]:
```suggestion
        /* In most cases, xPortStartScheduler() will not return. If it
         * returns pdTRUE then there was not enough heap memory available
         * to create either the Idle or the Timer task. If it returned
         * pdFALSE, then the application called xTaskEndScheduler().
         * Most ports don't implement xTaskEndScheduler() as there is
         * nothing to return to. */

```

File: include/task.h (hunk 1 above)
Comment (pierrenoel-bouteville-act) [reply include/task.h:1686]: How to do that ?

Comment (aggarg) [reply include/task.h:1686]: In general, you can put those changes in 2 separate branches and raise 2 separate PRs. However, malloc failed hook is not invoked from tasks.c and therefore, task.h does not seem the right place to put it. Can you just remove that change and keep only the  identical branches change?

Comment (aggarg) [reply include/task.h:1686]: Thank you. Would you please update the comment also as I suggested below?
